### DISCUSSION TITLE
Similar results not sure how to pick the best

### DISCUSSION THREAD

**Discussion by KevalRajpal-Digimonk** [osm-search/Nominatim#3388] [Geocoding around the world]
I am using *nominatim* `search` and `details` API. I have successfully setup on my own server as well. My problem is look at this search result
[Result](https://nominatim.openstreetmap.org/search?q=zvezdara&format=jsonv2&accept-language=en-US)

I am getting 3 records for the results, 2 of which are _exactly similar_. When I call `details` api with place id I am getting different polygon.

Q1. I want to get more details about these record and which I can consider to be my correct result and how I can tackle this problem for other place search results as well.
Q2. What purpose does the place_rank and importance serve?
Q3. I have added the serbia data in my own server, how I can update with other region of europe i.e. Italy, Montenegro and so on.

Thanks in advance.

**Comment by otbutz** [comment]
https://www.openstreetmap.org/relation/11514197 is a subarea of https://www.openstreetmap.org/relation/10625811 which also explains the higher `admin_level`.

I think the confusion comes from Nominatim adding the various `name` localizations from the parent relation:

https://nominatim.openstreetmap.org/ui/details.html?osmtype=R&osmid=11514197&class=boundary

![grafik](https://github.com/osm-search/Nominatim/assets/22315436/b3171cbe-571d-4ba9-8864-c245801a0e9b)

**Reply by KevalRajpal-Digimonk** [reply]
Hello @otbutz , Thanks for the quick response.

I can't find the `admin_level` in search result API.

```
{
    "place_id": 1233581,
    "licence": "Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright",
    "osm_type": "relation",
    "osm_id": 6930224,
    "lat": "44.7997558",
    "lon": "20.5032678",
    "category": "boundary",
    "type": "administrative",
    "place_rank": 16,
    "importance": 0.186676666666667,
    "addresstype": "suburb",
    "name": "Zvezdara Urban Municipality",
    "display_name": "Zvezdara Urban Municipality, City of Belgrade, Central Serbia, 11050, Serbia",
    "boundingbox": [
      "44.7280993",
      "44.8149904",
      "20.4774728",
      "20.5852472"
    ]
  }
  ```
  
  This is the single object from the API. For `admin_level` I need to have the run the `detail` API. How I can add the `admin_level` in my search API response.

**Reply by mtmail** [reply]
None of the /search output formats returns the admin_level. As approximation you can use place_rank divided by 2.

**Comment by mtmail** [comment]
I think You uncovered a data issue and the local community (can be found via https://openstreetmap.community/) would need to review the various admin levels in that city.

Both results https://www.openstreetmap.org/relation/6930224 and https://www.openstreetmap.org/relation/11514197 shared the same label (relation member). I removed the label from the suburb now but I think it needs reindexing of the area to show effect.

Additionally https://www.openstreetmap.org/relation/6930224 and https://www.openstreetmap.org/relation/10625811 seem to represent the same with different tags. Could be valid reason or mistake nobody noticed yet.

Nominatim search can't do much about duplicate result. It sorts (ranks) them by importance but both have the same importance score currently. It's better to fix the data issues. Such duplicate results are very rare, having the same importance even more.

The frontend you see on https://nominatim.openstreetmap.org/ui/search.html?q=Zvezdara is a separate project https://github.com/osm-search/nominatim-ui/ You can run it locally against your installation or install it on your server.

>  Q2. What purpose does the place_rank and importance serve?

Each place is put in a hierarchy in the Nominatim database. It starts with admin_level*2 but there can be exceptions and logic for individual countries. Admin_level=2 is a country, 4 a state, 8 a city, 10 and there's a huge table on https://wiki.openstreetmap.org/wiki/Tag%3aboundary=administrative for all countries. Putting the world into a hierarchy is never easy. For you to remember is: the lower the number the bigger the place.

Importance is calculated based on the place type (some defaults for each type) and how many wikipedia articles link to the place's wikipedia article. (Plus more logic related to different languages, normalization of titles, adding wikidata). Other geocoders also add population numbers.

>  Q3. I have added the serbia data in my own server, how I can update with other region of europe i.e. Italy, Montenegro and so on.

See https://nominatim.org/release-docs/latest/admin/Advanced-Installations/#importing-multiple-regions-with-updates If you have follow-up questions on multi-region imports it's best to open a new separate discussion.

**Comment by KevalRajpal-Digimonk** [comment]
Hello @mtmail, 
Thanks for sharing this valuable insight.

Do you suggest some patch for this problem; or some solution I can provide a fix in my project?
Can I use `place_rank` for this type of issue?
I will share more places where I identified this issue for betterment of community resources.

**Reply by mtmail** [reply]
In your project you can run

1. Database query

   ```sql
   DELETE FROM placex WHERE osm_type='R' AND osm_id in (10625811, 6930224);
   ```

2. nominatim CLI

   ```bash
   nominatim add-data --relation 10625811
   nominatim add-data --relation 6930224
   nominatim index
   ```

Then the results are (at least on my test server)

1. "Zvezdara Urban Municipality, Belgrade, City of Belgrade, Central Serbia, 11050, Serbia"
2. "Zvezdara, Belgrade, City of Belgrade, Central Serbia, 11060, Serbia" (the peak)
3. "Београд (Звездара), Zvezdara Urban Municipality, Belgrade, City of Belgrade, Central Serbia, 11050, Serbia"

The first and third result have the same boundary and many of the same tagging. It's not clear which is the better result. In that city there's full coverage of admin_level=8 and admin_level=9 (see https://overpass-turbo.eu/s/1JAw) so neither seems to be a mistake. I guess that's just how the city structures itself administratively.

Deleting and re-adding the place is not ideal, but `nominatim refresh` doesn't work here. The places "took" names from a linked nodes and there's no permanent data linking the relations back to the node in the Nominatim database.
